[Refactor] migrate Explorer home to FC and enforce type safety

Huxpro opened the pull request:
## Summary

This PR 
- migrated the Explorer home screen to Functional Components.
- properly configuring TSConfig and make uses of `<input>` element typed.
- add README to the projects.


## Checklist

<!--- Check and mark with an "x" -->

- [x] Tests updated (or not required).
- [x] Documentation updated (or not required).

Huxpro commented on explorer/homepage/index.tsx:
```diff
@@ -11,92 +11,89 @@ import SettingsPage from '@components/settingspage';
-    );
-  }
+  return (
+    <view clip-radius="true" style={{ height: '100%' }}>
```
@zhongyr is there a performance difference between inline styling via strings or objects?

zhongyr replied:
Objects performs better, no need to separate tokens into independent property and value KVs before parsing styles.

colinaaa replied:
There is no difference between an object literal and a string literal in this context. Both are converted into a `__AddInlineStyle` call:

```js
__AddInlineStyle(element, /* kHeight */ 25, '100%')
```

Huxpro commented on explorer/homepage/package.json:
```diff
@@ -2,22 +2,23 @@
     "@lynx-js/qrcode-rsbuild-plugin": "^0.3.3",
     "@lynx-js/react-rsbuild-plugin": "^0.9.8",
     "@lynx-js/rspeedy": "^0.9.3",
+    "@lynx-js/types": "^3.3.0",
```
Should have used workspace version if the type resolution works

Huxpro commented:
Does anyone have any clue why intrinsic elements e.g. `<view>` insisted to resolve to the one defined by `react` rather than the one we overrode to `@lynx-js/types`? cc @colinaaa @f0rdream 

In this project:

<img width="602" alt="image" src="https://github.com/user-attachments/assets/5b16443e-16c4-403d-a62e-6560ef270c7c" />

In `lynx-stack/examples/`:

<img width="511" alt="image" src="https://github.com/user-attachments/assets/f16a3977-d8b4-4d99-8cee-a308fc43c570" />

colinaaa commented:
> Does anyone have any clue why intrinsic elements e.g. `<view>` insisted to resolve to the one defined by `react` rather than the one we overrode to `@lynx-js/types`? cc @colinaaa @f0rdream 

Try `"jsx: "react-jsx"` with `"jsxImportSource": "@lynx-js/react"`. You may need to upgrade to `@lynx-js/react` v0.108.0. See https://github.com/lynx-family/lynx-stack/pull/545

Huxpro commented:
/land

jianliang00 commented:
🛬 Landing process started! >>553.0

jianliang00 commented:
🛬 Landing process started! >>554.0

Huxpro committed:
[Refactor] Migrate Explorer home to FC and enforce type safety

- migrated the Explorer home screen to Functional Components.
- properly configuring TSConfig and extend lynx-js/types.
- add README to the projects.

doc: https://github.com/lynx-family/lynx/tree/develop/explorer

Huxpro commented:
/land

jianliang00 commented:
🛬 Landing process started! >>557.0

jianliang00 commented:
🛬 Landing process started! >>558.0

Huxpro commented:
/land

jianliang00 commented:
🛬 Landing process started! >>559.0

jianliang00 commented:
🛬 Landing process started! >>560.0

Huxpro commented:
/land

jianliang00 commented:
🛬 Landing process started! >>561.0

jianliang00 commented:
🛬 Landing process started! >>562.0

Huxpro commented:
/land

jianliang00 commented:
🛬 Landing process started! >>563.0

jianliang00 commented:
🛬 Landing process started! >>564.0

Huxpro commented:
/land

jianliang00 commented:
🛬 Landing process started! >>565.0

jianliang00 commented:
🛬 Landing process started! >>566.0
